README.md

```diff
@@ -101,47 +101,50 @@ Spectral includes a number of ready made rules and functions for OpenAPI v2 and
 
 This example uses the OpenAPI v3 rules to lint a document.
 
-```javascript
+```ts
 const { Spectral } = require('@stoplight/spectral');
-const { oas3Functions, oas3Rules } = require('@stoplight/spectral/rulesets/oas3');
+const { oas3Functions, rules: oas3Rules } = require('@stoplight/spectral/rulesets/oas3');
 // for YAML
 const { parseWithPointers } = require("@stoplight/yaml");
-const myOAS = parseWithPointers(`
-responses:
-  '200':
-    description: ''
-    schema:
-      $ref: '#/definitions/error-response'
-`)
-
-// an OASv3 document
-const myOAS = {
-  // ... properties in your document
-  responses: {
-    '200': {
-      description: '',
-      schema: {
-        $ref: '#/definitions/error-response',
+
+// Using the anonymous async wrapper for the sake of the example.
+(async function() {
```

Comment:
AFAIR we wanted to avoid having an async IIFE in the documentation.

Reply:
Did we? I makes the whole example less readable imo

Reply:
Yeah, we did.
We had a discussion about examples when releasing 2.0
I don't remember the reasoning behind that decision anymore, but there was some.
@tbarn, do you still recall it?

Reply:
I like the IIFE version more as well, though.

Reply:
I like the IIFE version more as well, though.

Reply:
Looking at this example I am actually struggling to understand what is being done. Why don't we keep the README focusing on the CLI usage and show how easily you can get pretty or JSON errors back, then we can strip out all the JS Lib usage and make a new docs/js-api.md.

Reply:
@P0lip changed to plain promises

Reply:
@philsturgeon sure why not!

Reply:
In order to have a working example, it needed that async in there. Otherwise, it would just run as is.